File: HouseholdManager/Models/Recipes/Recipe.cs
using System.ComponentModel.DataAnnotations;

namespace HouseholdManager.Models.Recipes;

public class Recipe
{
    [Key] [Required] public string Name { get; set; } = null!;
    [Required] public List<RecipeIngredient> Ingredients { get; set; } = new();
    
    public string RecipeLink { get; set; } = String.Empty;
}

File: HouseholdManager/FrontEnd/src/data/Recipe.ts
export interface RecipeList {
  id: number
  householdId: number
  name: string
  recipes: Recipe[]
}

export interface Recipe {
  name: string
  recipeLink?: string
  ingredients: RecipeIngredient[]
}

export interface RecipeIngredient {
  id: number
  name: string
  amount: string
  price: number
}
